letter-sender/src/main/java/com/christmas/letter/sender/exceptions/LetterValidator.java

```diff
@@ -0,0 +1,21 @@
+package com.christmas.letter.sender.exceptions;
+
+import com.christmas.letter.sender.model.Letter;
+import java.util.regex.Pattern;
+import org.springframework.stereotype.Component;
+
+@Component
+public class LetterValidator{
+  private static final Pattern EMAIL_PATTERN = Pattern.compile(
+      "^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+\\.[a-zA-Z]{2,6}$");
+
+  public void validate(Letter letter) {
+
+    if (letter.getEmail() == null || !EMAIL_PATTERN.matcher(letter.getEmail()).matches()) {
```

Comment:
isn't the email format already being validated with the @Email annotation from the Letter class?

Reply:
yes just I will keep the validation but i do not use it